indica/workflows/bayes_workflow.py

```diff
@@ -0,0 +1,1065 @@
+
+    def process_data(self, model_callable: Callable):
+        self.pre_process_data(model_callable)
+        opt_data = flatdict.FlatDict(self.binned_data, ".")
+        if "xrcs.spectra" in opt_data.keys():
+            background = opt_data["xrcs.spectra"].where(
+                (opt_data["xrcs.spectra"].wavelength < 0.392)
+                & (opt_data["xrcs.spectra"].wavelength > 0.388),
+                drop=True,
+            )
+            opt_data["xrcs.spectra"]["error"] = np.sqrt(
+                opt_data["xrcs.spectra"] + background.std(dim="wavelength") ** 2
+            )
+        if "ts.ne" in opt_data.keys():
+            opt_data["ts.ne"]["error"] = opt_data["ts.ne"].max(dim="channel") * 0.05
+
+        if "ts.te" in opt_data.keys():
+            opt_data["ts.ne"]["error"] = opt_data["ts.ne"].max(dim="channel") * 0.05
+        self.opt_data = opt_data
+
+
+@dataclass
+class PhantomData(DataContext):
+    phantoms = True
+
+    def read_data(self, ):
+        self.reader = ReadST40(self.pulse, tstart=self.tstart, tend=self.tend, dt=self.dt, )
+        self.reader(self.diagnostics, revisions=self.reader_settings.revisions, filters=self.reader_settings.filters)
+        missing_keys = set(self.diagnostics) - set(self.reader.binned_data.keys())
+        if len(missing_keys) > 0:
+            raise ValueError(f"missing data: {missing_keys}")
+        self.equilibrium = self.reader.equilibrium
+        self.transforms = self.reader.transforms
+        self.raw_data = {}
+        self.binned_data = {}
+
+    def data_strategy(self):
+        print("Data strategy: Phantom data")
+        return self.model_callable()
+
+    def process_data(self, model_callable: Callable):
+        self.pre_process_data(model_callable)
+        self.opt_data = flatdict.FlatDict(self.binned_data, ".")
+
+
+@dataclass
+class MockData(PhantomData):
+    diagnostic_transforms: dict = field(default_factory=lambda: {"xrcs": helike_transform_example(1),
+                                                                 "smmh1": smmh1_transform_example(1),
+                                                                 "cxff_pi": pi_transform_example(5),
+                                                                 "cxff_tws_c": pi_transform_example(3),
+                                                                 "ts": ts_transform_example(11), })
+
+    def read_data(self):
+        print("Reading mock equilibrium / transforms")
+        self.equilibrium = fake_equilibrium(
+            tstart,
+            tend,
+            dt,
+        )
+        missing_transforms = list(set(diagnostics).difference(self.diagnostic_transforms.keys()))
+        if missing_transforms:
+            raise ValueError(f"Missing transforms: {missing_transforms}")
+
+        self.transforms = self.diagnostic_transforms
+        self.binned_data: dict = {}
+        self.raw_data: dict = {}
+
+
+@dataclass
+class OptimiserEmceeSettings:
```

Comment:
Why do we need a dataclass for this? 
In this application, what does having a class give us that a dictionary doesn't?